browser/src/map/Clipboard.js

```diff
@@ -828,26 +851,44 @@ L.Clipboard = L.Class.extend({
 			return false;
 		}
 
-		app.socket.sendMessage('uno ' + this._unoCommandForCopyCutPaste);
-		const url = this.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
+		const command = this._unoCommandForCopyCutPaste;
+		app.socket.sendMessage('uno ' + command);
+
+		// This is sent down the websocket URL which can race with the
+		// web fetch - so first step is to wait for the result of
+		// that command so we are sure the clipboard is set before
+		// fetching it.
+
 		const that = this;
+
+		if (that._commandCompletion.length > 0)
+			window.app.console.error('Already have ' + that._commandCompletion.length +
+						 ' pending clipboard command(s)');
+
+		const url = that.getMetaURL() + '&MimeType=text/html,text/plain;charset=utf-8';
+
+		// Share a single fetch
+		var fetchPromise = new Promise((resolve, reject) => {
+			try {
+				var result = fetch(url).then(response => response.text());
+				resolve(result);
+			} catch (err) {
+				reject(err);
+			}
+		});
+
+		var awaitPromise = function(url, mimetype, shorttype) {
+			return new Promise((resolve, reject) => {
```

Comment:
Oh nice, I didn't consider that the callback can be shared to fetch the data only once.

Reply:
=)

Reply:
Ah yes, this was my 1st thought at adding a new reply with a serial in it so we can undo any races related to which UNO command was invoked and which the reply is for - which would be by far more ideal, but is a bit vicious to get threaded through the core & API & so on ;-)